avro-serializer/src/test/java/io/confluent/kafka/serializers/KafkaAvroSerializerTest.java

```diff
@@ -849,4 +890,23 @@ public void testKafkaAvroSerializerReflectionRecordWithPrimitives() {
     assertEquals(message, obj);
   }
 
+  @Test
+  public void testKafkaAvroSerializerGenericRecordWithConverters() {
```

Comment:
Can you add tests for SpecificData and ReflectData as well?

Reply:
As far as I could see all tests in `AbstractKafkaAvroDeserializerTest` use a generic. I certainly can add a test for specific, that's quite straightforward, but I need to dig a little deeper for `ReflectData` since I've never used it myself.

Reply:
Ok my bad, they are all there. I started to write test for `SpecificData` and actually I wonder if setting the logical types converters is really needed since the maven plugin (properly configured) will do it anyway

Reply:
So after a big a further digging, this does not work as all for reflection because the this relies on the `getRecommendedSchema` method of each conversions and for example for Decimal the recommendation is `throw new UnsupportedOperationException("No recommended schema for decimal (scale is required)");`. So I feel like there is no way with the current version of Avro (1.10.11) to get it to work. I think this PR is still relevant, especially for people working with GenericRecords (like us), so maybe I could add a bit of logic to check which implementation is being configure and throw some kind of unsupported operation when `SCHEMA_REFLECTION_CONFIG==true`. For SpecificRecord I think need to double check if this adds any value compared to having the avro compiler adding them itself (with this PR, we just set them again basically, it doesn't hurt but not sure it brings value).
Waiting for your feedback, thanks for helping out.

Reply:
Ok, let's just target GenericData for now and skip SpecificData and ReflectData.

Reply:
So I removed it from those two container types but I haven't found any exception to throw or any logger available for usage. The description is verbose enough though, so maybe this is enough?